community/security/src/main/java/org/neo4j/server/security/auth/BasicAuthManager.java

```diff
@@ -130,14 +156,14 @@ public void setPassword( AuthSubject authSubject, String username, String passwo
         {
             throw new AuthorizationViolationException( "Invalid attempt to change the password for user " + username );
         }
-        if ( setPassword( username, password ) == null )
+        if ( setUserPassword( username, password ) == null )
         {
             throw new IllegalArgumentException( "User " + username + " does not exist" );
         }
     }
 
     @Override
-    public User setPassword( String username, String password ) throws IOException
+    public User setUserPassword( String username, String password ) throws IOException
```

Comment:
I think the API shouldn't return anything, but throw exceptions if it fails. But this is just an extraction of the original API as used by the REST UserService, and fixing up the API has been worked on in a separate branch where we actually extend the functionality of the auth manager. So the intention was not to change the old API here in order to minimise code changes to UserService on 3.0. 
But since we touched the API in this PR and this is now on 3.1 I see your point in also cleaning it up.